tests/integ/test_tf.py

```diff
@@ -87,6 +87,33 @@ def test_tf_async(sagemaker_session):
         print('predict result: {}'.format(result))
 
 
+def test_tf_with_requirements(sagemaker_session):
```

Comment:
This test should assert that the library BeautifulSoup4 is installed in the container.
You can do that by pip freezing and saving it as a file in opt/ml/model.
You can check later that the model has the file.

That being said, consider if this test is really necessary.

Reply:
what are your reasons for this test being unnecessary?

Reply:
We are testing that the user supplied requeriments are working in the container side.
Not all the containers features are tested in the client, only the main ones, to avoid long tests covering edge cases here. 
One possible option, is to change our current integration tests to cover your scenario as well, but I would suggest you to import a library that is required for training.
Another option is to add an additional integration test that adds value besides just the requeriment itself. A Keras model that loads a hdf5 file would be a great example.

Reply:
We are testing that the user supplied requeriments are working in the container side.
Not all the containers features are tested in the client, only the main ones, to avoid long tests covering edge cases here. 
One possible option, is to change our current integration tests to cover your scenario as well, but I would suggest you to import a library that is required for training.
Another option is to add an additional integration test that adds value besides just the requeriment itself. A Keras model that loads a hdf5 file would be a great example.